src/Button/button-group.scss

```diff
@@ -15,7 +15,7 @@
     &:focus,
     &:active,
     &.active {
-      z-index: 1;
+      z-index: 2;
```

Comment:
[inform] On hover on a non-active/focused button in the `ButtonGroup` would overlay the focus ring. This change ensures the focus ring stays above the hovered button's background.

Reply:
This change now lives in https://github.com/openedx/paragon/pull/3596